posydon/popsyn/binarypopulation.py

```diff
@@ -100,25 +100,47 @@ def __init__(self, **kwargs):
         self.population_properties = self.kwargs.get('population_properties',
                                                      SimulationProperties())
         atexit.register(lambda: BinaryPopulation.close(self))
-
+        self.metallicity = self.kwargs.get('metallicity', 1)
+        
+        # grab all metallicities in population or use single metallicity
+        self.metallicities = self.kwargs.get('metallicities', [self.metallicity])
+        
         self.population_properties.max_simulation_time = self.kwargs.get(
             'max_simulation_time')  # years
 
         entropy = self.kwargs.get('entropy', None)
         seq = np.random.SeedSequence(entropy=entropy)
-
+        
+        # Local MPI run
+        self.comm = self.kwargs.pop('comm', None)
+        # Job array ID runs
         self.JOB_ID = self.kwargs.pop('JOB_ID', None)
-        if self.JOB_ID is not None:
+        
+        if self.comm is not None and self.JOB_ID is not None:
+            raise ValueError('MPI and Job array runs are not compatible.')
+        # To guarantee reproducibility, we need to set the seed sequence
+        # to be the same across all processes.
+        elif self.comm is not None and self.entropy is None:
+            raise ValueError('A local MPI run requires an entropy value to be set.')
+        # local MPI run
+        elif self.comm is not None:
```

Comment:
Minor, just for readability of the code I suggest:
```
        # To guarantee reproducibility, we need to set the seed sequence
        # to be the same across all processes.
        elif self.comm is not None and self.entropy is None:
            raise ValueError('A local MPI run requires an entropy value to be set.')
        # local MPI run
        elif self.comm is not None:
            ...
```
should read:
```
        # local MPI run
        elif self.comm is not None:
            # To guarantee reproducibility, we need to set the seed sequence
            # to be the same across all processes.
            if self.entropy is None:
                raise ValueError('A local MPI run requires an entropy value to be set.')
            ...
```
because it is not another option of the outer decision tree. Usually, this should be a bit faster as well but may not in python.